src/main/scala/com/fulcrumgenomics/fasta/SequenceInfos.scala

```diff
@@ -0,0 +1,171 @@
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+package com.fulcrumgenomics.fasta
+
+import com.fulcrumgenomics.FgBioDef._
+import enumeratum.EnumEntry
+import htsjdk.samtools.{SAMSequenceDictionary, SAMSequenceRecord}
+
+
+/** Stores information about a single Sequence (ex. chromosome, contig)
+  *
+  * @param name the primary name of the sequence
+  * @param length the length of the sequence, or zero if unknown
+  * @param aliases the list of aliases for this sequence
+  * @param attributes attributes of this sequence
+  */
+case class SequenceInfo(name: String,
+                        length: Int,
+                        aliases: Seq[String] = Seq.empty,
+                        attributes: Map[String, Any] = Map.empty
+                       ) {
+
+  allNames.foreach { name => SAMSequenceRecord.validateSequenceName(name) }
+  require(length > 0, s"Length must be > 0 for '$name'")
+
+  /** All names, including aliases */
+  @inline final def allNames: Seq[String] = name +: aliases
+  @inline final def apply[T](key: String): T = this.attributes(key).asInstanceOf[T]
+  @inline final def get[T](key: String): Option[T] = this.attributes.get(key).map(_.asInstanceOf[T])
+  @inline final def contains(key: String): Boolean = this.attributes.contains(key)
+
+  @inline final def isAlternate: Boolean = this.contains("AH")
+  def alternate: Option[AlternateLocus] = {
+    this.apply[String]("AH") match {
+      case "*" => None
+      case string =>
+        val Array(refName, rest) = string.split(':')
+        val Array(start, end) = rest.split('-')
+        val locus = refName match {
+          case "=" => AlternateLocus(refName=this.name, start=start.toInt, end=end.toInt)
+          case _   => AlternateLocus(refName=refName, start=start.toInt, end=end.toInt)
+        }
+        Some(locus)
+    }
+  }
+
+  @inline final def m5: String = this.apply(SAMSequenceRecord.MD5_TAG)
```

Comment:
I would think anything beyond name and length should be `Option[T]` unless there is a sensible default value to sub in.